File: typescript/examples/leverage.ts
#!/usr/bin/env npx ts-node
// @ts-nocheck
/**
 * Leverage Example
 *
 * Update leverage for a position.
 *
 * Requires: PRIVATE_KEY environment variable
 */

import { HyperliquidSDK } from '@quicknode/hyperliquid-sdk';

const PRIVATE_KEY = process.env.PRIVATE_KEY;

if (!PRIVATE_KEY) {
  console.error("Set PRIVATE_KEY environment variable");
  console.error("Example: export PRIVATE_KEY='0x...'");
  process.exit(1);
}

async function main() {
  const sdk = new HyperliquidSDK(undefined, { privateKey: PRIVATE_KEY });
  console.log(`Wallet: ${sdk.address}`);

  // Update leverage for BTC to 10x cross margin
  const result = await sdk.updateLeverage("BTC", 10, true);
  console.log(`Update leverage result: ${JSON.stringify(result)}`);

  // Update leverage for ETH to 5x isolated margin
  // const result = await sdk.updateLeverage("ETH", 5, false);
  // console.log(`Update leverage result: ${JSON.stringify(result)}`);

  console.log("\nLeverage methods available:");
  console.log("  sdk.updateLeverage(asset, leverage, isCross)");
}

main().catch(console.error);

File: python/examples/leverage.py
#!/usr/bin/env python3
"""
Leverage Example

Update leverage for a position.

Requirements:
    pip install hyperliquid-sdk

Usage:
    export PRIVATE_KEY="0x..."
    python leverage.py
"""

import os
import sys

from hyperliquid_sdk import HyperliquidSDK

PRIVATE_KEY = os.environ.get("PRIVATE_KEY")

if not PRIVATE_KEY:
    print("Error: Set PRIVATE_KEY environment variable")
    print("  export PRIVATE_KEY='0xYourPrivateKey'")
    sys.exit(1)


def main():
    print("Hyperliquid Leverage Example")
    print("=" * 50)

    sdk = HyperliquidSDK(private_key=PRIVATE_KEY)
    print(f"Wallet: {sdk.address}")

    # Update leverage for BTC to 10x cross margin
    result = sdk.update_leverage("BTC", 10, is_cross=True)
    print(f"Update leverage result: {result}")

    # Update leverage for ETH to 5x isolated margin
    # result = sdk.update_leverage("ETH", 5, is_cross=False)
    # print(f"Update leverage result: {result}")

    print()
    print("Leverage methods available:")
    print("  sdk.update_leverage(asset, leverage, is_cross)")


if __name__ == "__main__":
    main()
